### PR TITLE
Add supported boundaries

### PR DESCRIPTION
**Pull request by tan-ro**

### CODE REVIEW

**Comment by nanux** [comment]
@tan-ro is this containing all the planned changes?

**Comment by tan-ro** [comment]
No, I've been working on it locally

**Comment by nanux** [comment]
Pushed updated filename to fix navigation.

It was throwing:
```
mkdocs_awesome_pages_plugin.navigation.NavEntryNotFound: Nav entry "SUPPORT_BOUNDARIES.md" not found. [/Users/abrokes/code/data-center-helm-charts/docs/docs/troubleshooting/.pages]
```

**Comment by nanux** [comment]
I've also added https://github.com/atlassian-labs/data-center-helm-charts/pull/236/commits/0e29bb837f16d25536e807c44ea56fd807c1d354 to prevent this in future.

File: docs/docs/README.md [hunk 1]
```diff
@@ -36,11 +36,11 @@ The diagram below provides a high level overview of what a typical deployment mi
 * [Configuration](userguide/CONFIGURATION.md) - a deep dive into the configuration parameters
 * [Platforms support](platforms/PLATFORMS.md) - how to allow support for different platforms
 * [Examples](examples/EXAMPLES.md) - various configuration examples
-* [Troubleshooting](userguide/TROUBLESHOOTING.md) - how to debug issues with installation
+* [Troubleshooting](troubleshooting/TROUBLESHOOTING.md) - how to debug issues with installation
 
 ## Feedback
 
-If you find any issue, please [raise a ticket](https://github.com/atlassian-labs/data-center-helm-charts/issues/new){.external}. If you have general feedback or question regarding the charts, please use [Atlassian Community Kubernetes space](https://community.atlassian.com/t5/Atlassian-Data-Center-on/gh-p/DC_Kubernetes){.external}.
+If you find any issue, [raise a ticket](https://github.com/atlassian-labs/data-center-helm-charts/issues/new){.external}. If you have general feedback or question regarding the charts, use [Atlassian Community Kubernetes space](https://community.atlassian.com/t5/Atlassian-Data-Center-on/gh-p/DC_Kubernetes){.external}.
```
Comment (errcode1202) [inline docs/docs/README.md:43]: Should this be:

`If you find any issues,` (plural)

also 

`or questions regarding` (plural)

File: docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Support boundaries
+
+This page describes what is within our scope of support for Kubernetes deployments, and what isn't. 
+
+## Supported components
+### Helm charts
+
+Helm is a Kubernetes package manager. It allows us to provide generic YAML templates that you configure with the specific values for your environments.
+
+You are responsible for creating the components that are required by the product for your type of deployment, as describes in the [Prerequisites](../userguide/PREREQUISITES.md). You need to insert the values (hostnames, configuration values) into your specific `values.yaml` file that is used for installation. We provide documentation for different configuration options in the [Configuration guide](../userguide/CONFIGURATION.md) and the [Examples](../examples/EXAMPLES.md).
```
Comment (errcode1202) [inline docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md:10]: The words:
`for your type of deployment, as describes`

should be:

`for your type of deployment, as described` 

replace `describes` with `described`

File: docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+If you have followed our documentation on how to configure the Helm charts, and you're using correctly created components, we will provide support if you encounter an error in installation. 
+
+If you find any issue, [raise a ticket](https://github.com/atlassian-labs/data-center-helm-charts/issues/new){.external}. If you have general feedback or questions regarding the charts, use [Atlassian Community Kubernetes space](https://community.atlassian.com/t5/Atlassian-Data-Center-on/gh-p/DC_Kubernetes){.external}.
```
Comment (errcode1202) [inline docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md:14]: pluralise `issue`?

File: docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+## Unsupported components
+There is a set of required components that you can create in multiple ways. You are responsible for creating them correctly so you can use them in the Helm charts.
```
Comment (errcode1202) [inline docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md:17]: Can we add a link to these components à la the prerequisites

File: docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+If you have followed our documentation on how to configure the Helm charts, and you're using correctly created components, we will provide support if you encounter an error in installation. 
+
+If you find any issue, [raise a ticket](https://github.com/atlassian-labs/data-center-helm-charts/issues/new){.external}. If you have general feedback or questions regarding the charts, use [Atlassian Community Kubernetes space](https://community.atlassian.com/t5/Atlassian-Data-Center-on/gh-p/DC_Kubernetes){.external}.
+
+## Unsupported components
+There is a set of required components that you can create in multiple ways. You are responsible for creating them correctly so you can use them in the Helm charts. 
+
+### Kubernetes cluster
+Helm charts need to be installed in the existing Kubernetes cluster. You need to make sure that you have enough privileges to run the application and create all the necessary entities that the Helm charts require. There are also different Kubernetes flavours that might require specific knowledge of how to install the products in them. For example, OpenShift and Rancher have more strict rules regarding container permissions.
```
Comment (errcode1202) [inline docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md:20]: Just wondering if the first sentence is required? i.e. `Helm charts need to be installed in the existing Kubernetes cluster.`

File: docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+See examples of [provisioning Kubernetes clusters on cloud-based providers](../examples/cluster/CLOUD_PROVIDERS.md).
+
+### Shared storage
+Kubernetes setup requires you to have shared storage if you want to have a clustered instance. It's completely up to you how you set up the shared storage. The main requirement is that this storage needs to be accessible from Kubernetes and needs to be accessible from multiple pods.
```
Comment (errcode1202) [inline docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md:25]: can hyperlink `shared storage` to our shared storage section in the configuration guide

File: docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+You can use a managed storage solution like EFS, Azure files, or some other dedicated solution that provides NFS-like access (e.g. dedicated NFS server, NetApp).
+
+Thereis a large number of combinations and potential setup scenarios and we can't support all of them. Our Helm charts expect you to provide a persistent volume claim or a similar accessible shared storage in the `values.yaml` file.
```
Comment (errcode1202) [inline docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md:29]: Typo `Thereis`

File: docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+### Database
+The database is provided as a connection string with the option to provide credentials and a driver. The database needs to be configured following product-specific requirements as per usual and needs to be accessible from Kubernetes.
+
+See an example of [provisioning databases on cloud-based providers](../examples/database/CLOUD_PROVIDERS.md).
+
+## Not supported configurations
+These configurations are explicitly not supported and the Helm charts don’t work without modifications in these environments:
+
+* Istio infrastructure
```
Comment (errcode1202) [inline docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md:49]: Can we hyperlink `Istio` to the official page

**Review by errcode1202 — CHANGES_REQUESTED** [review]

File: docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+It is up to you to make sure that the network configuration doesn’t prevent nodes from communicating with each other and other components.
+
+You also need to make sure that your instance is accessible to the users (DNS, firewalls, VPC config).
+
+### Database
+The database is provided as a connection string with the option to provide credentials and a driver. The database needs to be configured following product-specific requirements as per usual and needs to be accessible from Kubernetes.
+
+See an example of [provisioning databases on cloud-based providers](../examples/database/CLOUD_PROVIDERS.md).
+
+## Not supported configurations
```
Comment (errcode1202) [inline docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md:46]: Might be better to rename this section: `Unsupported configurations`

**Review by errcode1202 — APPROVED** [review]

File: docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+These configurations are explicitly not supported and the Helm charts don’t work without modifications in these environments:
+
+* Istio infrastructure
+    * Due to several reasons, Istio is imposing networking rules on every workload in the Kubernetes cluster that doesn’t work with our deployments.
```
[the remark this answers is not in the document]
Comment (tan-ro) [reply docs/docs/troubleshooting/SUPPORT_BOUNDARIES.md:51]: I missed that one. I'll fix it in the next PR

**Pull request merged**